File: app/src/main/java/com/kirill/todo/tasks/data/AbstractTask.java
package com.kirill.todo.tasks.data;

import java.io.Serializable;
import java.util.List;

public abstract class AbstractTask implements Serializable {

    protected String CREATOR;
    protected String taskName;
    protected int stepsCount;
    protected String description;
    protected TasksEnum type;
    protected boolean isActivated;
    protected String createdAt;
    protected int checkedWhen;

    protected List<String> whichDaysOfWeek;
    protected List<String> steps;

    public String getTaskName() {
        return taskName;
    }

    public void setTaskName(String taskName) {
        this.taskName = taskName;
    }

    public int getStepsCount() {
        return stepsCount;
    }

    public void setStepsCount(int stepsCount) {
        this.stepsCount = stepsCount;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public TasksEnum getType() {
        return type;
    }

    public void setType(TasksEnum type) {
        this.type = type;
    }

    public boolean isActivated() {
        return isActivated;
    }

    public void setActivated(boolean activated) {
        isActivated = activated;
    }

    public String isCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(String createdAt) {
        this.createdAt = createdAt;
    }

    public int getCheckedWhen() {
        return checkedWhen;
    }

    public void setCheckedWhen(int checkedWhen) {
        this.checkedWhen = checkedWhen;
    }

    public List<String> getWhichDaysOfWeek() {
        return whichDaysOfWeek;
    }

    public String getCreatedAt() {
        return createdAt;
    }

    public List<String> getSteps() {
        return steps;
    }
}

File: app/src/main/java/com/kirill/todo/tasks/data/AbstractTask.kt
package com.kirill.todo.tasks.data

import java.io.Serializable

abstract class AbstractTask : Serializable {

    private lateinit var createdAt: String
    private var whenActivated: Int = 0

    private var checked: Boolean = false
    private var whichDaysOfWeek: List<String> = mutableListOf()

    private var steps: List<String> = mutableListOf()

    protected lateinit var taskName: String
    protected lateinit var description: String
    protected lateinit var type: TasksEnum

    fun taskName(): String {
        return taskName
    }

    fun steps(): List<String> {
        return steps
    }

    fun steps(steps: List<String>) {
        this.steps = steps
    }

    fun taskName(taskName: String): Unit {
        this.taskName = taskName
    }

    fun description(): String {
        return description
    }

    fun description(description: String) {
        this.description = description
    }

    fun type(): TasksEnum {
        return type
    }

    fun type(type: TasksEnum) {
        this.type = type
    }

    fun whenActivated(): Int {
        return whenActivated
    }

    fun whenActivated(whenActivated: Int) {
        this.whenActivated = whenActivated
    }

    fun createdAt(createdAt: String) {
        this.createdAt = createdAt
    }

    fun createdAt(): String {
        return createdAt
    }

    fun whichDaysOfWeek(): List<String> {
        return whichDaysOfWeek
    }

    fun whichDaysOfWeek(newValue: List<String>): Unit {
        this.whichDaysOfWeek = newValue
    }

    fun checked(): Boolean {
        return checked
    }

    fun checked(newValue: Boolean): Unit {
        this.checked = newValue
    }
}
